src/scheduler.cc

```diff
@@ -31,26 +31,24 @@ Status SchedulerService::SubmitTask(ServerContext* context, const SubmitTaskRequ
       result_objrefs.push_back(result);
     }
     {
-      std::lock_guard<std::mutex> reference_counts_lock(reference_counts_lock_); // we grab this lock because increment_ref_count assumes it has been acquired
+      auto reference_counts_ptr = reference_counts_.get(); // we grab this lock because increment_ref_count assumes it has been acquired
       increment_ref_count(result_objrefs); // We increment once so the objrefs don't go out of scope before we reply to the worker that called SubmitTask. The corresponding decrement will happen in submit_task in raylib.
       increment_ref_count(result_objrefs); // We increment once so the objrefs don't go out of scope before the task is scheduled on the worker. The corresponding decrement will happen in deserialize_task in raylib.
     }
 
     auto operation = std::unique_ptr<Operation>(new Operation());
     operation->set_allocated_task(task.release());
     {
-      std::lock_guard<std::mutex> workers_lock(workers_lock_);
-      operation->set_creator_operationid(workers_[request->workerid()].current_task);
+      auto workers_ptr = workers_.get();
+      operation->set_creator_operationid((*workers_ptr)[request->workerid()].current_task);
     }
 
     OperationId operationid;
```

Comment:
remove curly braces below and put stuff in one line (OperationId operationid = ...)